### ISSUE TITLE
Retry support for Feign clients

### ISSUE THREAD

**Issue by Nikolas-Charalambidis** [resilience4j/resilience4j#1234]
Feign client is a very popular alternative to `RestTemplate` or newer REST-based clients.

As far as I see in the current guide website [Feign Guide for v1.6.1](https://resilience4j.readme.io/docs/feign), Resilience4j provides no support of the Retry module for Feign clients:

> **Current Features**
> - CircuitBreaker
> - RateLimiter
> - Fallback

<sub>I have found no issue discussing or proposing this.</sub>

**Comment by Romeh** [comment]
@Nikolas-Charalambidis  yes it is supported via FeignDecorators , for example of usage please check the unit test here : 

https://github.com/resilience4j/resilience4j/blob/9546fac42426e50517480507ec3ccae7ae268a49/resilience4j-feign/src/test/java/io/github/resilience4j/feign/Resilience4jFeignRetryTest.java

Resilience4jFeignRetryTest class has unit testing examples about the usage of retry .

**Comment by Nikolas-Charalambidis** [comment]
@Romeh Thanks for the reply and the sample. However, I am interested particularly in the annotations usage as long as I use Spring Boot, i.e. using `@Retry` as described at [Getting started with resilience4j-spring-boot2]( https://resilience4j.readme.io/docs/getting-started-3)

Does exist such support? If so, is an example of the usage available? Otherwise, is this feature planned?

**Comment by Romeh** [comment]
@Nikolas-Charalambidis  you can find an example how to use spring annotation with feign client here as well : 

https://github.com/resilience4j/resilience4j/blob/1f0e62ffffd651bd818ed3bbdf55369fea590b94/resilience4j-spring-boot2/src/test/java/io/github/resilience4j/service/test/retry/RetryDummyFeignClient.java

where the unit test for it in RetryAutoConfigurationTest  class

**Comment by Nikolas-Charalambidis** [comment]
That is what I was looking for! I wonder there is not the **retry** module mentioned at [Feign Guide for v1.6.1](https://resilience4j.readme.io/docs/feign) as the sample you provided looks like rather an experimental feature than the officially supported one. 

Shall I suggest an edit in that page through https://dash.readme.io/?

**Comment by RobWin** [comment]
Yes please. You are right. The documentation is not up-to-date.

**Comment by Nikolas-Charalambidis** [comment]
@RobWin I'll take care after that. Sadly, such edit is not related to the GitHub repository, so I suggest labeling this issue with the purple `documentation update` or anything more appropriate and I would inform you all here and close this issue once the edit is done and merged. What do you think?

**Comment by RobWin** [comment]
Fine for me
